### ISSUE TITLE
Fails To build on newset wayfire

### ISSUE THREAD

**Issue by CrezyDud** [WayfireWM/wayfire-plugins-extra#115]
output:
```
[14/43] Compiling C++ object src/libmag.so.p/mag.cpp.o
FAILED: src/libmag.so.p/mag.cpp.o
sccache c++ -Isrc/libmag.so.p -Isrc -I../src -I/usr/include/cairo -I/usr/include/lzo -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/include/sysprof-4 -I/usr/include/pixman-1 -I/usr/include/pango-1.0 -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/libdrm -I/usr/include/libevdev-1.0 -I/usr/include/libxml2 -flto=auto -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -Wnon-virtual-dtor -Wextra -std=c++17 -DWLR_USE_UNSTABLE -DWAYFIRE_PLUGIN -Wno-unused-parameter -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -fexceptions -Wp,-D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security -fstack-clash-protection -fcf-protection -Wp,-D_GLIBCXX_ASSERTIONS -fPIC -pthread -MD -MQ src/libmag.so.p/mag.cpp.o -MF src/libmag.so.p/mag.cpp.o.d -o src/libmag.so.p/mag.cpp.o -c ../src/mag.cpp
../src/mag.cpp: In lambda function:
../src/mag.cpp:176:30: error: ‘struct wlr_output’ has no member named ‘front_buffer’
  176 |         if (!output->handle->front_buffer)
      |                              ^~~~~~~~~~~~
../src/mag.cpp:184:52: error: ‘struct wlr_output’ has no member named ‘front_buffer’
  184 |         if (!wlr_buffer_get_dmabuf(output->handle->front_buffer, &dmabuf_attribs))
      |                                                    ^~~~~~~~~~~~
[30/43] Compiling C++ object src/libhinge.so.p/hinge.cpp.o
../src/hinge.cpp: In static member function ‘static void wayfire_hinge::send_message(wayfire_hinge::thread_message, int)’:
../src/hinge.cpp:107:14: warning: ignoring return value of ‘ssize_t write(int, const void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  107 |         write(write_fd, &message, 1);
      |         ~~~~~^~~~~~~~~~~~~~~~~~~~~~~
../src/hinge.cpp: In static member function ‘static int wayfire_hinge::on_pipe_update(int, uint32_t, void*)’:
../src/hinge.cpp:114:13: warning: ignoring return value of ‘ssize_t read(int, void*, size_t)’ declared with attribute ‘warn_unused_result’ [-Wunused-result]
  114 |         read(that->pipefd[0], &message, 1);
      |         ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
[31/43] Compiling C++ object src/libautorotate-iio.so.p/autorotate-iio.cpp.o
ninja: build stopped: subcommand failed.
==> ERROR: A failure occurred in build().
    Aborting...
 -> error making: wayfire-plugins-extra-git
```

**Comment by FriederHannenheim** [comment]
It looks like you're using the AUR package but I can confirm that the manual build is not working either.

**Comment by CrezyDud** [comment]
does using `source_back_buffer` instead of `output->handle->front_buffer` work?
something like that had been done in the update pr on wayfire:
![image](https://user-images.githubusercontent.com/84625713/152809537-fe484be2-0b57-4f33-a86a-9f398cf8b50e.png)

**Comment by NamorNiradnug** [comment]
> does using `source_back_buffer` instead of `output->handle->front_buffer` work? something like that had been done in the update pr on wayfire: ![image](https://user-images.githubusercontent.com/84625713/152809537-fe484be2-0b57-4f33-a86a-9f398cf8b50e.png)

no, still fails to build:
```
../src/mag.cpp: In lambda function:
../src/mag.cpp:176:14: error: ‘source_back_buffer’ was not declared in this scope
  176 |         if (!source_back_buffer)
      |              ^~~~~~~~~~~~~~~~~~
../src/mag.cpp:184:36: error: ‘source_back_buffer’ was not declared in this scope
  184 |         if (!wlr_buffer_get_dmabuf(source_back_buffer, &dmabuf_attribs))
      |                                    ^~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

**Comment by tombh** [comment]
@CrezyDud you were right, I've submitted exactly that fix in #118

**Comment by soreau** [comment]
Should be fixed with latest.
